File: 2149-remove-colored-pieces-if-both-neighbors-are-the-same-color/remove-colored-pieces-if-both-neighbors-are-the-same-color.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        countA = 0
        countB = 0
        consecutiveA = 0
        consecutiveB = 0

        for currentColor in colors:
            if currentColor == 'A':
                consecutiveA += 1
                consecutiveB = 0  # Reset consecutiveB count
            elif currentColor == 'B':
                consecutiveB += 1
                consecutiveA = 0  # Reset consecutiveA count
            else:
                # If it's not 'A' or 'B', reset both counts
                consecutiveA = 0
                consecutiveB = 0

            if consecutiveA >= 3:
                countA += 1

            if consecutiveB >= 3:
                countB += 1

        return countA > countB

File: 2149-remove-colored-pieces-if-both-neighbors-are-the-same-color/remove-colored-pieces-if-both-neighbors-are-the-same-color.java
class Solution {
    public boolean winnerOfGame(String colors) {
        StringBuffer s = new StringBuffer(colors);
        int countA = 0;
        int countB = 0;

        for (int i = 1; i < s.length() - 1; i++) {
            if (s.charAt(i) == 'A' && s.charAt(i - 1) == 'A' && s.charAt(i + 1) == 'A') {
                countA++;
            }
            if (s.charAt(i) == 'B' && s.charAt(i - 1) == 'B' && s.charAt(i + 1) == 'B') {
                countB++;
            }
        }

        return countA > countB;
    }
}
